src/customers/customers.controller.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import { CustomersService } from './customers.service';
+import { CreateCustomerDto } from '../dto/create-customer.dto';
+import { UpdateCustomerDto } from '../dto/update-customer.dto';
+
+@Controller('customers')
+export class CustomersController {
+  constructor(private readonly customersService: CustomersService) {}
+
+  @Post()
+  create(@Body() createCustomerDto: CreateCustomerDto) {
+    return this.customersService.create(createCustomerDto);
+  }
+
+  @Get()
+  index() {
+    return this.customersService.getAllCustomers();
+  }
+
+  @Get(':id')
+  show(@Param('id') id: string) {
+    return this.customersService.getCustomerById(+id);
+  }
+
+  @Put(':id')
+  update(
+    @Param('id') id: string,
+    @Body() updateCustomerDto: UpdateCustomerDto,
+  ) {
+    return this.customersService.updateCustomerById(+id, updateCustomerDto);
+  }
+
+  @Delete(':id')
+  remove(@Param('id') id: string) {
```

Comment:
By default NestJS returns 200, maybe it's worth hinting with 204 status code in the API that our DELETE does not return any resource object
```suggestion
  @Delete(':id')
  @HttpCode(204)
  remove(@Param('id') id: string) {
```